src/patient/patient-search.ts

```diff
@@ -1,255 +0,0 @@
-          <div class="flex-column">
-            <formly-form
-              [form]="form"
-              [fields]="fields"
-              [model]="model"
-              (modelChange)="syncInput($event)">
-            </formly-form>
-          </div>
-        </div>
-
-        <div class="d-flex flex-row gap-3" advanced>
-          <div class="flex-column">
-            <formly-form
-              [form]="form"
-              [fields]="advancedFields"
-              [model]="model"
-              (modelChange)="syncInput($event)">
-            </formly-form>
-          </div>
-        </div>
-      </app-generic>
-    </div>
-
-    <!-- Data table -->
-    <p-table
-      [value]="(this.searchService.data$ | async) || []"
-      dataKey="name"
-      [tableStyle]="{ 'min-width': '50rem' }"
-      [columns]="(searchService.columns$ | async) || []"
-      selectionMode="single"
-      (onRowSelect)="rowSelect($event)">
-      <ng-template pTemplate="header" let-columns>
-        <tr>
-          <th *ngFor="let col of columns">{{ col }}</th>
-        </tr>
-      </ng-template>
-      <ng-template pTemplate="body" let-rowData let-columns="columns">
-        <tr>
-          <td *ngFor="let col of columns">{{ rowData[col] }}</td>
-        </tr>
-      </ng-template>
-    </p-table>
-  </div>
-  `,
-})
-export class PatientSearchComponent {
-  searchStatus = ''; // for testing only
-  model: any; // needed for formly form
-
-  // status checkboxes not included in this form
-  // because they were just for testing form formatting
-  form: FormGroup = this.fb.group({
-    first_name: ['', [Validators.pattern('^[a-zA-Z]+$')]],
-    last_name: ['', [Validators.pattern('^[a-zA-Z]+$')]],
-    mrn: ['', [Validators.pattern('^[0-9]+$')]],
-  });
-  filterConfig = {
-    mrn: { field: 'MRN', connector: ' is ' },
-    first_name: { field: 'First Name', connector: ' starts with ' },
-    last_name: { field: 'Last Name', connector: ' starts with ' },
-  };
-
-  constructor(
-    public searchService: PatientSearchService,
-    private fb: FormBuilder,
-    private genericService: GenericService
-  ) { }
-
-  ngOnInit() {
-    this.genericService.form.set(this.form);
-    this.genericService.filterData.set(this.filterConfig);
-  }
-
-  syncInput(event: any) {
-    this.model = {
-      ...this.model,
-      first_name: event.first_name
-    }
-  }
-
-  search() {
-    this.searchStatus = `mrn: ${this.form.value.mrn}, firstName: ${this.form.value.first_name} lastName: ${this.form.value.last_name}`;
-    this.searchService.mockSearch();
-  }
-
-  rowSelect(row: any) {
-    console.log('rowSelected', row);
-  }
-
-  fields: FormlyFieldConfig[] = [
-    {
-      validators: {
-        validation: [atLeastOneRequired],
-      },
-      fieldGroup: [
-        {
-          key: 'first_name',
-          type: InputIconType,
-          props: {
-            label: 'First Name',
-            placeholder: 'First name',
-            error: 'Invalid First name',
-          },
```

Comment:
At scale, would we consider creating a registry of fields? That might allow for a simpler `fieldGroup` array value assignment and deduplication between the base and advanced fields.